### PR TITLE
Added ios support

### PR DESCRIPTION
**Pull request by KiritoDv**
This PR adds simple IOS support, it should not break any other platform since everything is wrapped on PLATFORM_IOS

### CODE REVIEW

File: extern/CMakeLists.txt [hunk 1]
```diff
@@ -16,14 +16,33 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Android")
...
+        FetchContent_Declare(
+            SDL2
+            GIT_REPOSITORY https://github.com/libsdl-org/SDL.git
+            GIT_TAG release-2.28.1
+    	)
+        message("SDL2 not found. Downloading now...")
+        FetchContent_MakeAvailable(SDL2)
+        message("SDL2 downloaded to " ${FETCHCONTENT_BASE_DIR}/sdl2-src)
+    endif()
+endif()
```
Comment (briaguya0) [inline extern/CMakeLists.txt]: this looks identical to the android one, why not just make the if for that include ios?

File: extern/CMakeLists.txt [hunk 2]
```diff
@@ -95,15 +114,15 @@ endif()
...
     target_include_directories(ImGui PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/metal-cpp)
     target_compile_definitions(ImGui PUBLIC IMGUI_IMPL_METAL_CPP)
 endif()
 
 if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
     target_link_libraries(ImGui PUBLIC SDL2::SDL2 SDL2::SDL2main)
-elseif (CMAKE_SYSTEM_NAME STREQUAL "CafeOS")
-    target_link_libraries(ImGui PUBLIC SDL2::SDL2-static)
+elseif (CMAKE_SYSTEM_NAME STREQUAL "CafeOS" OR CMAKE_SYSTEM_NAME STREQUAL "iOS")
+    target_link_libraries(ImGui PUBLIC SDL2::SDL2-static SDL2::SDL2main)
```
Comment (briaguya0) [inline extern/CMakeLists.txt]: why is this also changing what's being linked on wii u?

File: ios/Icon.png
Comment (briaguya0) [inline ios/Icon.png:1]: i feel like it probably makes sense for this to exist on the port side instead (so ports can have different icons)

File: ios/Launch.storyboard
Comment (briaguya0) [inline ios/Launch.storyboard:1]: just curious as to what this file does

File: ios/plist.in [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>CFBundleDevelopmentRegion</key>
+	<string>English</string>
+	<key>CFBundleDisplayName</key>
+	<string>Starship</string>
```
Comment (briaguya0) [inline ios/plist.in]: seeing this makes me think this part should probably be on the port side instead of the LUS side

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -512,16 +512,18 @@ target_link_libraries(libultraship
     PUBLIC ZAPDUtils ImGui tinyxml2 nlohmann_json::nlohmann_json
 )
 
+if (NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
```
Comment (briaguya0) [inline src/CMakeLists.txt]: this makes it so ios won't support o2r

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -550,22 +552,26 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
...
         $<$<NOT:$<CONFIG:Debug>>:SPDLOG_ACTIVE_LEVEL=1>
         $<$<BOOL:${USE_OPENGLES}>:USE_OPENGLES>
         $<$<BOOL:${GFX_DEBUG_DISASSEMBLER}>:GFX_DEBUG_DISASSEMBLER>
     )
+    if(NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
+        target_compile_definitions(libultraship PRIVATE
+            ENABLE_OPENGL
+        )
+    endif()
 else ()
```
Comment (briaguya0) [inline src/CMakeLists.txt]: kind of a nitpick, but i'd prefer this to not use nested logic

```
if not wii u
stuff that just doesn't happen on wii u
endif

if not wii u and not ios
stuff that doesn't happen on ios or wii u
endif
```

File: src/Context.cpp [hunk 6]
```diff
@@ -216,6 +216,9 @@ void Context::InitResourceManager(const std::vector<std::string>& otrFiles,
         SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_ERROR, "OTR file not found",
                                  "Main OTR file not found. Please generate one", nullptr);
         SPDLOG_ERROR("Main OTR file not found!");
+#ifdef PLATFORM_IOS
+        exit(0);
+#endif
```
Comment (briaguya0) [inline src/Context.cpp]: couple things

why `PLATFORM_IOS` as opposed to the patterns we have for other things like `__SWITCH__` - not a problem just curious

it'd be nice to add a comment explaining why exit 0 is needed for ios here

File: src/config/Config.h [hunk 7]
```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#ifdef __cplusplus
```
Comment (briaguya0) [inline src/config/Config.h]: why is this needed now/why isn't this a problem on other platforms?

File: src/graphic/Fast3D/gfx_metal.cpp [hunk 8]
```diff
@@ -186,12 +186,16 @@ static MTL::SamplerAddressMode gfx_cm_to_metal(uint32_t val) {
 // MARK: - ImGui & SDL Wrappers
 
 bool Metal_IsSupported() {
+#ifdef PLATFORM_IOS
+    return true;
+#else
```
Comment (briaguya0) [inline src/graphic/Fast3D/gfx_metal.cpp]: early return here means we don't need the else, can just endif

File: src/graphic/Fast3D/gfx_sdl2.cpp [hunk 9]
```diff
@@ -358,7 +358,11 @@ static void gfx_sdl_init(const char* game_name, const char* gfx_api_name, bool s
     height = window_height;
 #endif
 
+#ifdef PLATFORM_IOS
+    Uint32 flags = SDL_WINDOW_BORDERLESS | SDL_WINDOW_SHOWN;
+#else
     Uint32 flags = SDL_WINDOW_SHOWN | SDL_WINDOW_RESIZABLE | SDL_WINDOW_ALLOW_HIGHDPI;
+#endif
```
Comment (briaguya0) [inline src/graphic/Fast3D/gfx_sdl2.cpp]: any idea why ios needs this but android doesn't?

File: src/resource/archive/ArchiveManager.cpp [hunk 10]
```diff
@@ -5,7 +5,9 @@
 
 #include "resource/archive/Archive.h"
 #include "resource/archive/OtrArchive.h"
+#ifndef PLATFORM_IOS
 #include "resource/archive/O2rArchive.h"
+#endif
```
Comment (briaguya0) [inline src/resource/archive/ArchiveManager.cpp]: still curious about the ios not supporting libzip thing

File: src/window/Window.cpp [hunk 11]
```diff
@@ -47,6 +47,7 @@ Window::~Window() {
 void Window::Init() {
     bool steamDeckGameMode = false;
     bool androidGameMode = false;
+    bool iosGameMode = false;
```
Comment (briaguya0) [inline src/window/Window.cpp]: do we use these separately at all? seems like we could probably just use a single "gameMode" variable and set it appropriately based on platform

File: src/window/Window.cpp [hunk 12]
```diff
@@ -237,18 +242,18 @@ void Window::InitWindowManager() {
             mWindowManagerApi = &gfx_dxgi_api;
             break;
 #endif
-#if defined(ENABLE_OPENGL) || defined(__APPLE__)
+#ifdef ENABLE_OPENGL
```
Comment (briaguya0) [inline src/window/Window.cpp:234]: i'm curious as to why this was an or in the first place, is `ENABLE_OPENGL` not set on mac?

File: src/window/gui/Gui.cpp [hunk 13]
```diff
@@ -210,6 +210,7 @@ void Gui::ImGuiBackendInit() {
             ImGui_ImplGX2_Init();
             break;
 #else
+#ifdef ENABLE_OPENGL
```
Comment (briaguya0) [inline src/window/gui/Gui.cpp:214]: having this right after the else from wii u makes for a bit of complex nesting, might be nice to add some comments on the endifs to make it easier to tell where these start/end

File: src/window/gui/Gui.cpp [hunk 14]
```diff
@@ -720,10 +722,12 @@ void Gui::ImGuiRenderDrawData(ImDrawData* data) {
             ImGui_ImplWiiU_DrawKeyboardOverlay();
             break;
 #else
+#ifdef ENABLE_OPENGL
```
Comment (briaguya0) [inline src/window/gui/Gui.cpp:728]: i think this one can be elif instead of nested

**Review by briaguya0 — COMMENTED** [review]
overall the stuff that seems similar to the android stuff seems reasonable, there are a few things in here that feel like they're better off living on the port side than the LUS side though

File: ios/plist.in (hunk 3 above)
Comment (KiritoDv) [reply ios/plist.in]: This is replaced on the CMakeLists from the port

File: src/graphic/Fast3D/gfx_sdl2.cpp (hunk 9 above)
Comment (KiritoDv) [reply src/graphic/Fast3D/gfx_sdl2.cpp]: SDL_WINDOW_ALLOW_HIGHDPI is broken on ios rn

File: src/resource/archive/ArchiveManager.cpp (hunk 10 above)
Comment (KiritoDv) [reply src/resource/archive/ArchiveManager.cpp]: I tried for hours building libzip but i didnt manage to do it

File: src/window/Window.cpp (hunk 12 above)
Comment (KiritoDv) [reply src/window/Window.cpp:234]: On mac ENABLE_OPENGL could be optional, but in that ifdef it was forced on

File: cmake/ios.toolchain.cmake
Comment (briaguya0) [inline cmake/ios.toolchain.cmake:1]: This file should not be added to LUS. We can recommend it in the readme and link to https://github.com/leetal/ios-cmake, ports can include it, and for CI we can wget it, but it should not be checked in to LUS.

If we check it in to LUS that means:
* People will try to PR changes to it
* Ports will use it, meaning pulling in upstream changes might break ports - coupling the iOS cmake toolchain version to the LUS version (something that should stay decoupled)

File: cmake/dependencies/ios.cmake [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+endif()
+
+#=================== nlohmann-json ===================
+FetchContent_Declare(
+    nlohmann_json
+    GIT_REPOSITORY https://github.com/nlohmann/json.git
+    GIT_TAG v3.11.3
+    OVERRIDE_FIND_PACKAGE
+)
+message("nlohmann_json not found. Downloading now...")
```
Comment (briaguya0) [inline cmake/dependencies/ios.cmake]: This will say not found every time since it isn't in an if statement

File: cmake/dependencies/ios.cmake [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+FetchContent_MakeAvailable(nlohmann_json)
```
Comment (briaguya0) [inline cmake/dependencies/ios.cmake]: This should not be needed since we're calling `find_package()` later

from https://cmake.org/cmake/help/latest/module/FetchContent.html#integrating-with-find-package
> ```cmake
> include(FetchContent)
> FetchContent_Declare(
>   googletest
>   GIT_REPOSITORY https://github.com/google/googletest.git
>   GIT_TAG        703bd9caab50b139428cea1aaff9974ebee5742e # release-1.10.0
>   OVERRIDE_FIND_PACKAGE
> )
> FetchContent_Declare(
>   Catch2
>   GIT_REPOSITORY https://github.com/catchorg/Catch2.git
>   GIT_TAG        605a34765aa5d5ecbf476b4598a862ada971b0cc # v3.0.1
>   OVERRIDE_FIND_PACKAGE
> )
> 
> # The following will automatically forward through to FetchContent_MakeAvailable()
> find_package(googletest)
> find_package(Catch2)
> ```
key part there is
> The following will automatically forward through to FetchContent_MakeAvailable()

File: CMakeLists.txt [hunk 17]
```diff
@@ -14,15 +17,18 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
...
 include(cmake/dependencies/common.cmake)
 if (CMAKE_SYSTEM_NAME STREQUAL "Android")
 	include(cmake/dependencies/android.cmake)
-endif()
-if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
-	include(cmake/dependencies/windows.cmake)
+elseif (CMAKE_SYSTEM_NAME STREQUAL "iOS")
+    set(PLATFORM "OS64")
+    include(cmake/dependencies/ios.cmake)
+    include(cmake/ios.toolchain.cmake)
```
Comment (briaguya0) [inline CMakeLists.txt]: why is this being included instead of doing it the way we do with devkitpro?

```
-DCMAKE_TOOLCHAIN_FILE=/opt/devkitpro/cmake/Switch.cmake
```

File: CMakeLists.txt [hunk 18]
```diff
@@ -1,9 +1,12 @@
 cmake_minimum_required(VERSION 3.24.0)
+set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
```
Comment (briaguya0) [inline CMakeLists.txt]: based on this https://cmake.org/cmake/help/latest/policy/CMP0077.html it seems the `NEW` behavior has been the default for a while, what happens if this is removed?

File: CMakeLists.txt [hunk 19]
```diff
@@ -1,9 +1,12 @@
...
 
 option(NON_PORTABLE "Build a non-portable version" OFF)
+option(SIGN_LIBRARY "Enable xcode signing" OFF)
+option(BUNDLE_ID "Bundle ID for xcode signing" "com.example.libultraship")
```
Comment (briaguya0) [inline CMakeLists.txt]: can we put options behind an `if()`?

File: CMakeLists.txt [hunk 20]
```diff
@@ -14,15 +17,18 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
...
+set(ADDITIONAL_LIB_INCLUDES "")
 
 include(cmake/dependencies/common.cmake)
 if (CMAKE_SYSTEM_NAME STREQUAL "Android")
 	include(cmake/dependencies/android.cmake)
-endif()
-if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
-	include(cmake/dependencies/windows.cmake)
+elseif (CMAKE_SYSTEM_NAME STREQUAL "iOS")
+    set(PLATFORM "OS64")
```
Comment (briaguya0) [inline CMakeLists.txt]: my goal with this section was to *only* have dependency related stuff here. other platform specific settings can go to a different part of this file

File: CMakeLists.txt [hunk 21]
```diff
@@ -14,15 +17,18 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
...
 include(cmake/Utils.cmake)
+set(ADDITIONAL_LIB_INCLUDES "")
 
 include(cmake/dependencies/common.cmake)
 if (CMAKE_SYSTEM_NAME STREQUAL "Android")
 	include(cmake/dependencies/android.cmake)
-endif()
-if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
-	include(cmake/dependencies/windows.cmake)
+elseif (CMAKE_SYSTEM_NAME STREQUAL "iOS")
```
Comment (briaguya0) [inline CMakeLists.txt:28]: i intentionally didn't use `elseif` here. that way if we wanted to add something like
```
if (SOME_OPTIONAL_SET_OF_DEPS_FOR_MULTIPLE_PLATFORMS)
    include(cmake/dependencies/optionalset.cmake)
endif()
```
we could put that somewhere that makes sense in the block without needing to break up elses

File: extern/CMakeLists.txt [hunk 22]
```diff
@@ -2,7 +2,7 @@ find_package(OpenGL QUIET)
 
 if (CMAKE_SYSTEM_NAME MATCHES "Windows")
     find_package(SDL2 CONFIG REQUIRED)
-elseif (NOT CMAKE_SYSTEM_NAME STREQUAL "Android")
+elseif (NOT CMAKE_SYSTEM_NAME STREQUAL "Android" AND NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
```
Comment (briaguya0) [inline extern/CMakeLists.txt]: with `OVERRIDE_FIND_PACKAGE` i don't think this is needed

File: extern/CMakeLists.txt [hunk 23]
```diff
@@ -77,13 +77,15 @@ endif()
...
-if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+if (CMAKE_SYSTEM_NAME STREQUAL "Darwin" OR CMAKE_SYSTEM_NAME STREQUAL "iOS")
     target_include_directories(ImGui PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/metal-cpp)
     target_compile_definitions(ImGui PUBLIC IMGUI_IMPL_METAL_CPP)
 endif()
 
 if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
     target_link_libraries(ImGui PUBLIC SDL2::SDL2 SDL2::SDL2main)
+elseif (CMAKE_SYSTEM_NAME STREQUAL "iOS")
+    target_link_libraries(ImGui PUBLIC SDL2::SDL2-static SDL2::SDL2main)
```
Comment (briaguya0) [inline extern/CMakeLists.txt:88]: just curious, why do we need static and main here?

File: src/CMakeLists.txt [hunk 24]
```diff
@@ -101,7 +101,7 @@ target_sources(libultraship PRIVATE ${Source_Files__Window__Gui})
 
 file(GLOB Source_Files__Utils RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "utils/*.h" "utils/*.cpp")
 
-if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+if (CMAKE_SYSTEM_NAME STREQUAL "Darwin" OR CMAKE_SYSTEM_NAME STREQUAL "iOS")
```
Comment (briaguya0) [inline src/CMakeLists.txt:104]: doesn't need to happen in this PR but worth noting/making an issue about - we should probably rename `OSXFolderManager` for multiple reasons now
* `macOS` is just called `macOS` not `OSX`
* `iOS` isn't `OSX`

maybe just "AppleFolderManager" or something

File: src/CMakeLists.txt [hunk 25]
```diff
@@ -133,8 +133,13 @@ elseif (CMAKE_SYSTEM_NAME STREQUAL "NintendoSwitch")
...
     set(Source_Files__Port
-        ${CMAKE_CURRENT_SOURCE_DIR}/port/android/AndroidImpl.h
-        ${CMAKE_CURRENT_SOURCE_DIR}/port/android/AndroidImpl.cpp
+        ${CMAKE_CURRENT_SOURCE_DIR}/port/mobile/MobileImpl.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/port/mobile/MobileImpl.cpp
+    )
+elseif (CMAKE_SYSTEM_NAME STREQUAL "iOS")
+    set(Source_Files__Port
+        ${CMAKE_CURRENT_SOURCE_DIR}/port/mobile/MobileImpl.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/port/mobile/MobileImpl.cpp
```
Comment (briaguya0) [inline src/CMakeLists.txt]: can just use `OR` here

File: src/config/Config.h [hunk 26]
```diff
@@ -98,4 +98,4 @@ class Config {
     bool mIsNewInstance;
     std::map<uint32_t, std::shared_ptr<ConfigVersionUpdater>> mVersionUpdaters;
 };
-} // namespace LUS
+} // namespace LUS
```
Comment (briaguya0) [inline src/config/Config.h]: not having a newline at the end of this file is making it show up as part of the diff when it hasn't been changed. it'll be nice to clean this up for a clean diff

```suggestion
} // namespace LUS

```

File: src/graphic/Fast3D/gfx_metal.cpp [hunk 27]
```diff
@@ -186,12 +186,16 @@ static MTL::SamplerAddressMode gfx_cm_to_metal(uint32_t val) {
...
+#ifdef __IOS__
+    return true;
+#else
     NS::Array* devices = MTLCopyAllDevices();
     NS::UInteger count = devices->count();
 
     devices->release();
 
     return count > 0;
+#endif
```
Comment (briaguya0) [inline src/graphic/Fast3D/gfx_metal.cpp:199]: no need to wrap the rest of this in an else since this is an early return

it'd also be nice to add a comment explaining why the logic in here doesn't work as-is on iOS

File: src/resource/archive/O2rArchive.cpp [hunk 28]
```diff
@@ -83,4 +83,4 @@ bool O2rArchive::Close() {
 
     return true;
 }
-} // namespace LUS
+} // namespace LUS
```
Comment (briaguya0) [inline src/resource/archive/O2rArchive.cpp]: same thing as config.h

```suggestion
} // namespace LUS

```

File: src/resource/archive/O2rArchive.h [hunk 29]
```diff
@@ -1,5 +1,4 @@
 #pragma once
-
 #undef _DLL
```
Comment (briaguya0) [inline src/resource/archive/O2rArchive.h]: just to keep the diff clean we should put that newline back

also (def out of scope for this PR) we should make an issue to clean up `#pragma once` use and standardize on `#ifndef` based header guards

File: src/window/Window.cpp [hunk 30]
```diff
@@ -45,8 +45,7 @@ Window::~Window() {
 }
 
 void Window::Init() {
-    bool steamDeckGameMode = false;
-    bool androidGameMode = false;
+    bool mGameMode = false;
```
Comment (briaguya0) [inline src/window/Window.cpp:48]: out of scope for this PR and also mostly a @Kenix3 question but this has always felt a bit like a kitchen sink variable to me

 i'm thinking we should make an issue to find all usages of `mGameMode` and figure out either
* what a good name for it would be
or
* if we want to split it up into multiple variables to handle different things it changes

File: src/window/gui/Gui.cpp [hunk 31]
```diff
@@ -719,20 +717,20 @@ void Gui::ImGuiRenderDrawData(ImDrawData* data) {
             GX2SetScissor(0, 0, mImGuiIo->DisplaySize.x, mImGuiIo->DisplaySize.y);
             ImGui_ImplWiiU_DrawKeyboardOverlay();
             break;
-#else
+#elif defined(ENABLE_OPENGL)
```
Comment (briaguya0) [inline src/window/gui/Gui.cpp]: i'm having a hard time knowing where these blocks end when reviewing because of all the elifs, it's not the worst here (`ImGuiBackendInit` is rough) but if we instead had
```cpp
#ifdef __WIIU__
// gx2
#endif

#ifdef ENABLE_OPENGL
// sdl opengl
#endif
```
we could use whitespace to improve readability quite a bit

File: src/window/gui/Gui.cpp [hunk 32]
```diff
@@ -719,20 +717,20 @@ void Gui::ImGuiRenderDrawData(ImDrawData* data) {
...
-        case WindowBackend::SDL_METAL:
-            Metal_RenderDrawData(data);
-            break;
-#endif
+
 #if defined(ENABLE_DX11) || defined(ENABLE_DX12)
         case WindowBackend::DX11:
             ImGui_ImplDX11_RenderDrawData(data);
             break;
+#elif defined(__APPLE__)
```
Comment (briaguya0) [inline src/window/gui/Gui.cpp]: why is this being moved?

**Review by briaguya0 — COMMENTED** [review]
mostly small things, anything bigger i've noted as something that'd be nice to make an issue for and out of scope for this pr

File: src/graphic/Fast3D/gfx_metal.cpp (hunk 27 above)
Comment (KiritoDv) [reply src/graphic/Fast3D/gfx_metal.cpp:199]: I did it, but it didnt worked

File: CMakeLists.txt [hunk 33]
```diff
@@ -14,15 +20,32 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
...
 endif()
-if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
-	include(cmake/dependencies/windows.cmake)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "iOS")
+    set(PLATFORM "OS64COMBINED")
+    include(FetchContent)
+    FetchContent_Declare(iostoolchain
+        GIT_REPOSITORY https://github.com/leetal/ios-cmake
+        GIT_TAG origin/master
```
Comment (briaguya0) [inline CMakeLists.txt]: pinning latest feels a little risky here, maybe pin by sha?

File: extern/CMakeLists.txt (hunk 23 above)
Comment (briaguya0) [reply extern/CMakeLists.txt:88]: still curious about this one

File: src/CMakeLists.txt (hunk 24 above)
Comment (briaguya0) [reply src/CMakeLists.txt:104]: https://github.com/Kenix3/libultraship/issues/524

File: src/graphic/Fast3D/gfx_metal.cpp (hunk 27 above)
Comment (briaguya0) [reply src/graphic/Fast3D/gfx_metal.cpp:199]: ah, ios probably can't find the symbol

File: src/window/Window.cpp (hunk 30 above)
Comment (briaguya0) [reply src/window/Window.cpp:48]: https://github.com/Kenix3/libultraship/issues/525

**Review by briaguya0 — APPROVED** [review]

File: extern/CMakeLists.txt (hunk 23 above)
Comment (briaguya0) [reply extern/CMakeLists.txt:88]: from discord
> It was because SDLMain helps creating the needed ios views automatically

**Comment by briaguya0** [comment]
verified by https://github.com/HarbourMasters/Shipwright/pull/4076
